sklearn/ensemble/tests/test_forest.py

```diff
@@ -244,9 +244,9 @@ def test_poisson_vs_mse():
         # As squared_error might correctly predict 0 in train set, its train
         # score can be better than Poisson. This is no longer the case for the
         # test set. But keep the above comment for clipping in mind.
-        if val == "test":
+        if data_name == "val":
```

Comment:
I tried to remove this condition to see if the next assertion would also hold on the training set after the fix in #22191 but this is not the case.

Reply:
I checked this, too.